sootup.core/src/main/java/sootup/core/validation/TypesValidator.java

```diff
@@ -34,18 +39,48 @@ public class TypesValidator implements BodyValidator {
 
   @Override
   public List<ValidationException> validate(Body body, View view) {
-    /*
-     * SootMethod methodRef = body.getMethod();
-     *
-     * if (methodRef != null) { if (!methodRef.getReturnType().isAllowedInFinalCode()) { exceptions.add(new
-     * ValidationException(methodRef, "Return type not allowed in final code: " + methodRef.getReturnType(),
-     * "return type not allowed in final code:" + methodRef.getReturnType() + "\n methodRef: " + methodRef)); } for (Type t :
-     * methodRef.getParameterTypes()) { if (!t.isAllowedInFinalCode()) { exceptions.add(new ValidationException(methodRef,
-     * "Parameter type not allowed in final code: " + t, "parameter type not allowed in final code:" + t + "\n methodRef: " +
-     * methodRef)); } } } for (Local l : body.getLocals()) { Type t = l.getType(); if (!t.isAllowedInFinalCode()) {
-     * exceptions.add(new ValidationException(l, "Local type not allowed in final code: " + t, "(" + methodRef +
-     * ") local type not allowed in final code: " + t + " local: " + l)); } }
-     */
-    return null;
+    List<ValidationException> validationExceptions = new ArrayList<>();
+
+    Optional<? extends SootMethod> sootMethodOpt = view.getMethod(body.getMethodSignature());
+    if (!sootMethodOpt.isPresent()) {
+      throw new IllegalStateException(
+          "We should find the given method to the given Body in the View. wrong View or Method?");
+    }
+    SootMethod method = sootMethodOpt.get();
+
+    if (!(method.getReturnType() instanceof PrimitiveType
+            || method.getReturnType() instanceof VoidType
+            || method.getReturnType() instanceof ReferenceType)
+        || method.getReturnType() instanceof NullType) {
```

Comment:
nulltype should not be allowed (as a method definition can not have the null reference as definition - it is returned for a referencetype - is a different thing)